### DISCUSSION TITLE
depth testing in eframe paint callback with wgpu feature

### DISCUSSION THREAD

**Discussion by bananaturtlesandwich** [emilk/egui#3270] [Q&A]
Hi just what the title says. I've tried enabling depth testing in my paint callback by setting depth buffer to not be 0 in native options and setting depth stencil to be
```rust
depth_stencil: Some(wgpu::DepthStencilState {
    format: TextureFormat::Depth32Float,
    depth_write_enabled: true,
    depth_compare: CompareFunction::Less,
    stencil: Default::default(),
    bias: Default::default(),
}),
```
in my render pipeline but it fails to draw with `Incompatible depth-stencil attachment format: the RenderPass uses a texture with format None but the RenderPipeline uses an attachment with format Some(Depth32Float)`
Is there something I'm missing? please let me know if there is and if not I can try and make a PR to fix or switch to egui-wgpu and egui-winit

**Comment by bananaturtlesandwich** [comment]
I'm going to assume this isn't a thing so I'm going to move fully to wgpu

**Comment by emilk** [comment]
Are you using `eframe`? Did you set the `depth_buffer` and `stencil_buffer` fields in `NativeOptions` (https://docs.rs/eframe/latest/eframe/struct.NativeOptions.html#structfield.depth_buffer) ?

**Reply by bananaturtlesandwich** [reply]
already ported to new backend but code is here - the depth buffer is set to 1 since I believe the wgpu backend only checks if it's not 0 https://github.com/bananaturtlesandwich/stove/tree/2b6dba21ea4ef336b0a20dd4739620a76a0f3b44

**Reply by bananaturtlesandwich** [reply]
ah it's not there I remember trying it and it didn't work so I didn't commit but I did set depth buffer - I can make a branch with that if you want
